File: WebApp/indoor-localization/src/data-access/mock-repositories/MockFacilityRepository.ts
import { Facility } from "../../entities/Facility";
import { IFacilityRepository } from "../repository-interfaces/IFacilityRepository";

export class MockFacilityRepository implements IFacilityRepository {
  private static facilities: Facility[] = [];

  async Get(id: number): Promise<Facility | null> {
      return MockFacilityRepository.facilities.find(facility => facility.id === id) || null;
  }

  async GetAll(): Promise<Facility[]> {
      return MockFacilityRepository.facilities;
  }

  async Add(facility: Facility): Promise<boolean> {
      MockFacilityRepository.facilities.push(facility);
      return true;
  }

  async Delete(id: number): Promise<boolean> {
      const index = MockFacilityRepository.facilities.findIndex(facility => facility.id === id);
      if (index !== -1) {
          MockFacilityRepository.facilities.splice(index, 1);
          return true;
      }
      return false;
  }

  async Update(updatedFacility: Facility): Promise<boolean> {
      const index = MockFacilityRepository.facilities.findIndex(facility => facility.id === updatedFacility.id);
      if (index !== -1) {
          MockFacilityRepository.facilities[index] = updatedFacility;
          return true;
      }
      return false;
  }
}
  

File: Android/indoor_localization/lib/mock-repositories/mock_facility_repository.dart
import 'package:indoor_localization/entities/facility.dart';
import 'package:indoor_localization/repository-interfaces/facility_repository.dart';

class MockFacilityRepository implements IFacilityRepository {
  static final List<Facility> _facilities = [];

  @override
  Facility? get(int id) {
    return _facilities.firstWhere((facility) => facility.id == id);
  }

  @override
  List<Facility> getAll() {
    return _facilities;
  }

  @override
  bool add(Facility facility) {
    _facilities.add(facility);
    return true;
  }

  @override
  bool delete(int id) {
    final index = _facilities.indexWhere((facility) => facility.id == id);
    if (index != -1) {
      _facilities.removeAt(index);
      return true;
    }
    return false;
  }

  @override
  bool update(Facility updatedFacility) {
    final index = _facilities.indexWhere((facility) => facility.id == updatedFacility.id);
    if (index != -1) {
      _facilities[index] = updatedFacility;
      return true;
    }
    return false;
  }
}
